### ISSUE TITLE
regression: can't find crate

### ISSUE THREAD

**Issue by BoxyUwU** [rust-lang/rust#128892]
- https://crater-reports.s3.amazonaws.com/beta-1.81-2/beta-2024-07-26/gh/6d7a.rasn-compiler/log.txt

```
[INFO] [stdout] error[E0463]: can't find crate for `rasn_compiler_derive`
[INFO] [stdout]   --> /opt/rustwide/workdir/rasn-compiler-tests/src/helpers/mod.rs:14:13
[INFO] [stdout]    |
[INFO] [stdout] 14 |             rasn_compiler_derive::asn1!($asn1);
[INFO] [stdout]    |             ^^^^^^^^^^^^^^^^^^^^ can't find crate
```

**Comment by ehuss** [comment]
I have a very strong suspicion this is due to this warning:

```
INFO] [stderr] warning: output filename collision.
[INFO] [stderr] The lib target `rasn_compiler` in package `rasn-compiler v0.3.0 (/opt/rustwide/workdir/rasn-compiler)` has the same output filename as the lib target `rasn_compiler` in package `rasn-compiler v0.3.0 (/opt/rustwide/workdir/rasn-compiler)`.
[INFO] [stderr] Colliding filename is: /opt/rustwide/target/debug/deps/librasn_compiler.so
[INFO] [stderr] The targets should have unique names.
[INFO] [stderr] Consider changing their names to be unique or compiling them separately.
```

Because this workspace defines rasn-compiler as a cdylib, and rasn-compiler is used as both a regular and proc-macro dependency, and it is using cargo's `resolver="2"`, the cdylib ends up getting built twice in parallel. One rustc process will stomp over the `.so` while the other is running.

This is a known bug in cargo, tracked in https://github.com/rust-lang/cargo/issues/6313.

I'm not entirely certain why this is exhibiting as `rasn_compiler_derive` not found. I would not expect the `.so` filename collision to affect a proc-macro since the proc-macro should be loading the `.rlib`, not the `.so` (and the `.rlib` has a unique filename).

I cannot reproduce the `can't find crate` error on any version. I suspect this is not a regression.

**Comment by Mark-Simulacrum** [comment]
I'm going go ahead and close as we think this isn't a regression.
